### ISSUE TITLE
Add delay period to tendermint client

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#7133]
## Summary

Add a [delayPeriod](https://github.com/cosmos/ibc-go/blob/0bf8e8bc6611b5940ad455470f22f8c274feb783/proto/ibc/core/connection/v1/connection.proto#L31) to the tendermint client state

## Problem Definition

With eureka, the connection layer is removed entirely. If we want to support the ability for clients to have delay periods to increase security, we should add it to the tendermint client (any other clients which would like to add this feature should also add it)

## Proposal

Add the delay period to the tendermint client. Set it to 0 by default (this should allow the client state encoding to remain unchanged for backwards compatibility - soon shouldn't be a consideration with removal of self client/consensus state validation). 

In Eureka, setting the client to a non-zero delay period would be no problem and give support for delay periods. 

In classic, it would be a problem, only if the two chains haven't updated to: v7.8+, v8.5+, v9+. To remedy and support a removal of the connection layer entirely, we should:
* add ability to modify delay period on a client via gov #1171 

Thus, if connections have non zero delay periods, gov can choose to update or leave as is for the underlying client before we remove connections and thus create the same fundamental security layer for connections sharing the same client. 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by lacsomot** [comment]
Hi, i created a PR for this. Can you help me review it? TYSM

**Comment by DimitrisJim** [comment]
Hey! Sorry but this issue needed some more discussion by the team before being picked up. Will add icebox to it for now and close related PR.

Ideally might be able to pick it up after it is discussed! Thanks and sorry again for the confusion.

**Comment by DimitrisJim** [comment]
Removing `eureka` tag, this can be tackled after feature lands.

**Comment by womensrights** [comment]
I don't think we should add the delay period to the tendermint light client given that it is an unused feature, therefore closing this issue

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
